cores/esp8266/IPAddress.cpp

```diff
@@ -77,39 +96,72 @@ bool IPAddress::fromString(const char *address) {
         // Too few dots (there must be 3 dots)
         return false;
     }
-    _address.bytes[3] = acc;
+    (*this)[3] = acc;
+
+    setV4();
     return true;
 }
 
 IPAddress& IPAddress::operator=(const uint8_t *address) {
-    memcpy(_address.bytes, address, sizeof(_address.bytes));
+    setV4();
+    v4() = *reinterpret_cast<const uint32_t*>(address);
     return *this;
 }
 
 IPAddress& IPAddress::operator=(uint32_t address) {
-    _address.dword = address;
+    setV4();
+    v4() = address;
     return *this;
 }
 
 bool IPAddress::operator==(const uint8_t* addr) const {
-    return memcmp(addr, _address.bytes, sizeof(_address.bytes)) == 0;
+    return isV4() && v4() == *reinterpret_cast<const uint32_t*>(addr);
 }
 
 size_t IPAddress::printTo(Print& p) const {
     size_t n = 0;
-    for(int i = 0; i < 3; i++) {
-        n += p.print(_address.bytes[i], DEC);
-        n += p.print('.');
+
+    if (!isSet())
+        return p.print(F("(IP unset)"));
```

Comment:
What is the point reasoning of this? Lots of strange behavior reported, Not sure where to start investigating, but this is showing up in places